pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go

```diff
@@ -1,328 +1,160 @@
-		return "", "", "", "", err
+		return nil, err
 	}
 
-	obj, err := data.Convert(machine)
+	obj, err := data.Convert(infraMachine)
 	if err != nil {
-		return "", "", "", "", err
+		return nil, err
 	}
 
 	if capiMachine.Spec.InfrastructureRef.APIVersion == "rke-machine.cattle.io/v1" {
-		if obj.String("status", "jobName") == "" {
-			return corev1.ConditionUnknown, "NoJob", "waiting to schedule machine create", "", nil
-		}
+		if capiMachine.DeletionTimestamp.IsZero() {
+			if obj.String("status", "jobName") == "" {
+				return &machineStatus{
+					cond:    Provisioned,
+					status:  corev1.ConditionUnknown,
+					reason:  "NoJob",
+					message: "waiting to schedule machine create",
+				}, nil
+			}
 
-		if !obj.Bool("status", "jobComplete") {
-			return corev1.ConditionUnknown, "Creating",
-				fmt.Sprintf("creating server (%s) in infrastructure provider", capiMachine.Spec.InfrastructureRef.Kind), "", nil
-		}
-	}
+			if !obj.Bool("status", "jobComplete") {
+				return &machineStatus{
+					cond:    Provisioned,
+					status:  corev1.ConditionUnknown,
+					reason:  "Creating",
+					message: fmt.Sprintf("creating server (%s) in infrastructure provider", capiMachine.Spec.InfrastructureRef.Kind),
+				}, nil
+			}
+		} else {
+			if obj.String("status", "failureReason") == string(capierror.DeleteMachineError) {
+				return &machineStatus{
+					cond:   InfrastructureReady,
+					status: corev1.ConditionFalse,
+					reason: capi.DeletionFailedReason,
+					message: fmt.Sprintf("failed deleting server (%s) in infrastructure provider: %s: %s",
+						capiMachine.Spec.InfrastructureRef.Kind,
+						obj.String("status", "failureReason"),
+						obj.String("status", "failureMessage"),
+					),
+					providerID: obj.String("spec", "providerID"),
+				}, nil
+			}
 
-	if capiMachine.DeletionTimestamp != nil && obj.String("status", "failureReason") == string(capierror.DeleteMachineError) {
-		capiMachine.Status.FailureReason = &[]capierror.MachineStatusError{capierror.DeleteMachineError}[0]
-		return corev1.ConditionFalse, capi.DeletionFailedReason,
-			fmt.Sprintf("failed deleting server (%s) in infrastructure provider: %s: %s",
-				capiMachine.Spec.InfrastructureRef.Kind,
-				obj.String("status", "failureReason"),
-				obj.String("status", "failureMessage")), obj.String("spec", "providerID"), nil
+			return &machineStatus{
+				cond:       InfrastructureReady,
+				status:     corev1.ConditionUnknown,
+				reason:     capi.DeletingReason,
+				message:    fmt.Sprintf("deleting server (%s) in infrastructure provider", capiMachine.Spec.InfrastructureRef.Kind),
+				providerID: obj.String("spec", "providerID"),
+			}, nil
+		}
 	}
 
-	return "", "", "", obj.String("spec", "providerID"), nil
-}
-
-type dbinfo struct {
-	Members []member `json:"members,omitempty"`
-}
-type member struct {
-	ClientURLs []string `json:"clientURLs,omitempty"`
+	return &machineStatus{cond: Provisioned, providerID: obj.String("spec", "providerID")}, nil
```

Comment:
Should we be setting `status` to `corev1.ConditionTrue`?

Reply:
No, because if the status is empty here, then [here](https://github.com/rancher/rancher/pull/35403/files#diff-dbeeb0a3acefea09a1c49792078d38eeab261888ec9f34170f24fac122376f4bR208) Rancher will get the status from the plan.

Reply:
OK, thanks!